demo/site/src/layout/header/DekstopMenu.tsx

```diff
@@ -7,10 +7,10 @@ import FocusLock from "react-focus-lock";
 import { useIntl } from "react-intl";
 import styled from "styled-components";
 
-import { type GQLHeaderFragment } from "./Header.fragment.generated";
+import { type GQLDesktopMenuFragment } from "./DesktopMenu.fragment.generated";
 
 interface Props {
-    header: GQLHeaderFragment;
+    header: GQLDesktopMenuFragment;
```

Comment:
I would name the prop "menu" or "menuData".

And defining fragments is not enough, you also have to actually use them by importing them into headerFragment. And removing everything from headerFragment that you don't use in the header component.